subsys/metrics/src/main/java/org/commonjava/indy/metrics/zabbix/api/IndyZabbixApi.java

```diff
@@ -0,0 +1,299 @@
+package org.commonjava.indy.metrics.zabbix.api;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.apache.http.HttpEntity;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+/**
+ * Created by xiabai on 3/31/17.
+ */
+public class IndyZabbixApi
+                implements ZabbixApi
+{
+    private static final Logger logger = LoggerFactory.getLogger( IndyZabbixApi.class );
+
+    private CloseableHttpClient httpClient;
```

Comment:
You should `@Inject IndyHttpProvider` here, then use the `createClient()` method as needed below. This will give us a central place to handle SSL configuration and things like that.